### PR TITLE
Unexport all possible items

### PR DESCRIPTION
**Pull request by codysoyland**
## Overview

This branch is the result of a script that unexports everything possible while still passing tests.

This PR replaces #1455 and #1460.

## Pull request checklist

- [ ] I have read the [contributing guide](https://github.com/pilosa/pilosa/blob/master/CONTRIBUTING.md).
- [ ] I have agreed to the [Contributor License Agreement](https://cla-assistant.io/pilosa/pilosa).
- [ ] I have updated the [documentation](https://github.com/pilosa/pilosa/tree/master/docs).
- [ ] I have resolved any merge conflicts.
- [ ] I have included tests that cover my changes.
- [ ] All new and existing tests pass.

## Code review checklist
This is the checklist that the reviewer will follow while reviewing your pull request. You do not need to do anything with this checklist, but be aware of what the reviewer will be looking for.

- [ ] Ensure that any changes to external docs have been included in this pull request.
- [ ] If the changes require that minor/major versions need to be updated, tag the PR appropriately.
- [ ] Ensure the new code is [properly commented](https://github.com/golang/go/wiki/CodeReviewComments#doc-comments) and follows [Idiomatic Go](https://dmitri.shuralyov.com/idiomatic-go).
- [ ] Check that tests have been written and that they cover the new functionality.
- [ ] Run tests and ensure they pass.
- [ ] Build and run the code, performing any applicable integration testing.

### CODE REVIEW

File: cluster.go [hunk 1]
```diff
@@ -36,16 +36,16 @@ import (
...
 const (
-	// DefaultPartitionN is the default number of partitions in a cluster.
-	DefaultPartitionN = 256
+	// defaultPartitionN is the default number of partitions in a cluster.
+	defaultPartitionN = 256
 
 	// ClusterState represents the state returned in the /status endpoint.
-	ClusterStateStarting = "STARTING"
+	clusterStateStarting = "STARTING"
 	ClusterStateNormal   = "NORMAL"
```
Comment (jaffee) [inline cluster.go]: this is a bit weird - would it be terrible difficult to unexport this as well?

File: cluster.go [hunk 2]
```diff
@@ -36,16 +36,16 @@ import (
...
 
 const (
-	// DefaultPartitionN is the default number of partitions in a cluster.
-	DefaultPartitionN = 256
+	// defaultPartitionN is the default number of partitions in a cluster.
+	defaultPartitionN = 256
 
 	// ClusterState represents the state returned in the /status endpoint.
-	ClusterStateStarting = "STARTING"
+	clusterStateStarting = "STARTING"
```
Comment (travisturner) [inline cluster.go]: seems odd that these are partially exported

Comment (travisturner) [reply cluster.go]: by that I mean:
```
clusterStateStarting
ClusterStateNormal (exported)
clusterStateResizing
```

File: http/client.go [hunk 3]
```diff
@@ -60,7 +60,7 @@ func NewInternalClientFromURI(defaultURI *pilosa.URI, remoteClient *http.Client)
 	return &InternalClient{
 		defaultURI: defaultURI,
 		serializer: proto.Serializer{},
-		HTTPClient: remoteClient,
+		hTTPClient: remoteClient,
```
Comment (jaffee) [inline http/client.go]: should be httpClient

File: pql/pql.peg.go [hunk 4]
```diff
@@ -287,7 +287,7 @@ func (t *tokens32) Tokens() []token32 {
 	return t.tree
 }
 
-type PQL struct {
+type pql struct {
```
Comment (jaffee) [inline pql/pql.peg.go]: this is generated code and shouldn't be messed with... maybe we should hide it in an internal dir though

**Pull request merged**